File: my_py_pkg/my_py_pkg/number_publisher.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64


class NumberPublisher(Node):
    def __init__(self):
        super().__init__('number_publisher')
        self.number_ = 1
        self.number_publisher_ = self.create_publisher(
            msg_type=Int64, topic='number', qos_profile=10)
        self.number_timer_ = self.create_timer(
            timer_period_sec=1.0, callback=self.publish_number)
        self.get_logger().info('Number publisher has been started')

    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.number_publisher_.publish(msg=msg)
        self.get_logger().info(f'Number: {self.number_}')
        # self.number_ += 1


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node=node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/number_publisher.cpp
#include "example_interfaces/msg/int64.hpp"
#include "rclcpp/rclcpp.hpp"

class NumberPublisherNode : public rclcpp::Node {
public:
    NumberPublisherNode() : Node("number_publisher") {
        number_ = 1;
        number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        number_timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                                std::bind(&NumberPublisherNode::publishNumber, this));
        RCLCPP_INFO(this->get_logger(), "Number publisher has been started.");
    }

    void publishNumber() {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        number_publisher_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Number: %d", number_);
        // number_++;
    }

private:
    int number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::TimerBase::SharedPtr number_timer_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
